### PR TITLE
PHP Deprecated: ltrim(): Passing null to parameter #59154

### PR DESCRIPTION
**Pull request by NiharRan**
I added a check before filtering the link using the esc_url function.

function next_posts( $max_page = 0, $display = true ) {
	$link = get_next_posts_page_link( $max_page );

	$output = '';
	if ( $link ) {
		$output = esc_url( $link );
	}

	if ( $display ) {
		echo $output;
	} else {
		return $output;
	}
}

Trac ticket: 59154

### CODE REVIEW

**Review by Rajinsharwar — APPROVED** [review]
LGTM!

**Comment by codersantosh** [comment]
@NiharRan Ultra-rapid fix 😃

File: src/wp-includes/link-template.php [hunk 1]
```diff
@@ -2511,7 +2511,12 @@ function get_next_posts_page_link( $max_page = 0 ) {
  * @return string|void The link URL for next posts page if `$display = false`.
  */
 function next_posts( $max_page = 0, $display = true ) {
-	$output = esc_url( get_next_posts_page_link( $max_page ) );
+	$link = get_next_posts_page_link( $max_page );
+
+	$output = '';
+	if ( $link ) {
+		$output = esc_url( $link );
+	}
```
Comment (mukeshpanchal27) [inline src/wp-includes/link-template.php]:
```suggestion
	$link   = get_next_posts_page_link( $max_page );
	$output = $link ? esc_url( $link ) : '';
```

Use ternary operator for simplify it.

**Comment by hellofromtonya** [comment]
The change LGTM 👍 But it needs a test. I'll push a test to this PR shortly.

File: src/wp-includes/link-template.php (hunk 1 above)
Comment (hellofromtonya) [reply src/wp-includes/link-template.php]: I think either approach is fine. I'm not opinionated about either way. Both ways exist in Core.

**Comment by hellofromtonya** [comment]
Seems the service is hiccuping

>Pulling wordpress-develop (nginx:alpine)...
received unexpected HTTP status: 503 Service Unavailable

**Comment by hellofromtonya** [comment]
Running the new test locally on PHP 8.1 using the following command:
```
npm run test:php -- --filter Tests_Link_NextPosts
```

Before the fix, the test result:

```
1) Tests_Link_NextPosts::test_should_return_empty_string_when_no_next_posts_page_link
ltrim(): Passing null to parameter #1 ($string) of type string is deprecated

/var/www/src/wp-includes/formatting.php:4494
/var/www/src/wp-includes/link-template.php:2516
/var/www/tests/phpunit/tests/link/nextPost.php:37
```

After the fix, the test result:

```
PHPUnit 9.6.13 by Sebastian Bergmann and contributors.

Warning:       Your XML configuration validates against a deprecated schema.
Suggestion:    Migrate your XML configuration using "--migrate-configuration"!

.                                                                   1 / 1 (100%)

Time: 00:00.328, Memory: 166.50 MB
```

One of the first things I consider is if the fix breaks BC (backward compatibility). In other words, is the result (except for the deprecation) the same? This is why I forced pushed the test to come before the fix, i.e. to test what the result is.

Before the fix, the result is:

* an empty string is returned.
* On  PHP 8.1 and newer, a deprecation is triggered.

After the fix, an empty string is still returned.

This can also be seen in action here across different PHP versions https://3v4l.org/7cIvT. Notice, all versions return an empty string, even on PHP 8.1+.

This means: no BC break 🎉

**Review by hellofromtonya — APPROVED** [review]
[As noted here](https://github.com/WordPress/wordpress-develop/pull/5045#issuecomment-1739970893), the fix does indeed resolve the issue while maintaining BC of still returning an empty string.

LGTM 👍 Ready for commit.

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/56740. Thank you everyone for your contributions :)

**Pull request closed without merging**